tests/reporting/test_market_unit.py

```diff
@@ -0,0 +1,527 @@
+    payoutThridHashValue = initializeMarket.derivePayoutDistributionHash(payoutThridNumerators, False)
+    payoutFourthHashValue = initializeMarket.derivePayoutDistributionHash(payoutFourthNumerators, False)
+
+    push_to_designated_report_state(localFixture, mockUniverse, chain, initializeMarket, mockStakeToken, payoutDesignatedNumerators, mockStakeTokenFactory, mockReportingWindow, mockReputationToken, constants)
+    assert initializeMarket.getTentativeWinningPayoutDistributionHash() == initializeMarket.derivePayoutDistributionHash(payoutDesignatedNumerators, False)
+    
+    mockReportingWindow.reset()
+    # get second token
+    mockSecondStakeToken = set_mock_stake_token_value(localFixture, initializeMarket, payoutSecondNumeratorsDispute, mockStakeTokenFactory)
+    
+    # verify stake tokens are set
+    assert initializeMarket.getStakeToken(payoutDesignatedNumerators, False) == mockStakeToken.address
+    assert initializeMarket.getStakeToken(payoutSecondNumeratorsDispute, False) == mockSecondStakeToken.address
+
+    with raises(TransactionFailed, message="tentative winning payout distro can not be 0 we need stake token with positive supply"):
+        mockSecondStakeToken.callUpdateTentativeWinningPayoutDistributionHash(initializeMarket.address, payoutSecondHashDisputeValue)
+    
+    mockStakeToken.setTotalSupply(105)
+    mockSecondStakeToken.setTotalSupply(20)
+    assert initializeMarket.getTotalStake() == 0
+    assert mockSecondStakeToken.callIncreaseTotalStake(initializeMarket.address, 20)
+    assert initializeMarket.getTotalStake() == 20
+    assert mockReportingWindow.getIncreaseTotalStakeCalled() == True
+
+    assert mockSecondStakeToken.callUpdateTentativeWinningPayoutDistributionHash(initializeMarket.address, payoutSecondHashDisputeValue)
+    assert initializeMarket.getTentativeWinningPayoutDistributionHash() == initializeMarket.derivePayoutDistributionHash(payoutDesignatedNumerators, False)
+    assert initializeMarket.getBestGuessSecondPlaceTentativeWinningPayoutDistributionHash() == initializeMarket.derivePayoutDistributionHash(payoutSecondNumeratorsDispute, False)
+    # buy a thrid stake token
+    mockThridStakeToken = set_mock_stake_token_value(localFixture, initializeMarket, payoutThridNumerators, mockStakeTokenFactory)
+    mockThridStakeToken.setTotalSupply(200)    
+    # verify stake tokens have correct value
+    assert initializeMarket.getPayoutDistributionHashStake(payoutDesignatedHashValue) == 105
+    assert initializeMarket.getPayoutDistributionHashStake(payoutSecondHashDisputeValue) == 20
+    assert initializeMarket.getPayoutDistributionHashStake(payoutThridHashValue) == 200
+
+    assert mockThridStakeToken.callUpdateTentativeWinningPayoutDistributionHash(initializeMarket.address, payoutThridHashValue)
+    assert initializeMarket.getTentativeWinningPayoutDistributionHash() == initializeMarket.derivePayoutDistributionHash(payoutThridNumerators, False)
```

Comment:
I think rather than doing derive calls through the test you can just use the already derived hashes from above. Like in this case `payoutThirdHashValue`